js/livecoin.js

```diff
@@ -377,49 +378,101 @@ module.exports = class livecoin extends Exchange {
+            if (typeof timestamp !== 'undefined') {
+                if (timestamp.indexOf ('T') >= 0) {
+                    timestamp = this.parse8601 (timestamp);
+                } else {
+                    timestamp = this.safeInteger (order, 'lastModificationTime');
+                }
+            }
+        }
+        if (timestamp) {
+            datetime = this.iso8601 (timestamp);
+        }
         let trades = undefined;
         if ('trades' in order)
             // TODO currently not supported by livecoin
             // trades = this.parseTrades (order['trades'], market, since, limit);
             trades = undefined;
         let status = undefined;
-        if (order['orderStatus'] === 'OPEN' || order['orderStatus'] === 'PARTIALLY_FILLED') {
+        let orderStatus = undefined;
+        if ('status' in order) {
+            orderStatus = this.safeString (order, 'status');
+        }
+        if ('orderStatus' in order) {
+            orderStatus = this.safeString (order, 'orderStatus');
+        }
+        if (orderStatus === 'OPEN' || orderStatus === 'PARTIALLY_FILLED') {
             status = 'open';
-        } else if (order['orderStatus'] === 'EXECUTED' || order['orderStatus'] === 'PARTIALLY_FILLED_AND_CANCELLED') {
+        } else if (orderStatus === 'EXECUTED' || orderStatus === 'PARTIALLY_FILLED_AND_CANCELLED') {
             status = 'closed';
         } else {
             status = 'canceled';
         }
-        let symbol = order['currencyPair'];
-        let parts = symbol.split ('/');
-        let quote = parts[1];
+        let symbol = undefined;
+        if ('currencyPair' in order) {
+            symbol = order['currencyPair'];
+        }
+        if ('symbol' in order) {
+            symbol = order['symbol'];
+        }
+        let parts = undefined;
+        let quote = undefined;
+        if (symbol) {
+            parts = symbol.split ('/');
+            quote = parts[1];
+        }
+        // originally:
         // let [ base, quote ] = symbol.split ('/');
         let type = undefined;
         let side = undefined;
-        if (order['type'].indexOf ('MARKET') >= 0) {
-            type = 'market';
-        } else {
-            type = 'limit';
+        if ('type' in order) {
+            if (order['type'].indexOf ('MARKET') >= 0) {
+                type = 'market';
+            }
+            if (order['type'].indexOf ('LIMIT') >= 0) {
+                type = 'limit';
+            }
+            if (order['type'].indexOf ('SELL') >= 0) {
+                side = 'sell';
+            }
+            if (order['type'].indexOf ('BUY') >= 0) {
+                side = 'buy';
+            }
         }
-        if (order['type'].indexOf ('SELL') >= 0) {
-            side = 'sell';
-        } else {
-            side = 'buy';
+        let price = this.safeFloat (order, 'price');
+        let cost = this.safeFloat (order, 'commissionByTrade'); // commission_rate?
```

Comment:
The `cost` of an order is:
- `if (status === 'open' and filled === 0) { amount * price }`
- `if (status === 'closed' || status === 'canceled') { filled * price }`

The `cost` of a trade is `amount * price`.

So, an order `cost` or a trade `cost` means the total *quote* volume of the order (whereas `amount` is the *base* volume). The `cost` field itself is there mostly for convenience and can be deduced from other fields.

I'll add that to the Manual )

As for the unified `fee` sub-structure, it should be like this:

```
'fee': {
    'cost': 123.45,
    'rate': 0.001,
    'currency': 'ETH',
}
```